### PR TITLE
[RFC] Cleanup: Use @staticmethod

### PR DESCRIPTION
**Pull request by dwaynebailey**
Some of these might not be valid, but a few are.  We should mark things as `@staticmethod` if they are indeed that.  I'm not sure how much of a perf boost is involved but it certainly makes it clear.  Some of what I've selected may in fact not be valid thus the RFC.

### CODE REVIEW

File: pootle/apps/contact/forms.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ def from_email(self):
             self.cleaned_data['email']
         )
 
-    def recipient_list(self):
+    @staticmethod
```
Comment (unho) [inline pootle/apps/contact/forms.py]: This is extending a class that had this as a regular method. So undo this change.

File: pootle/apps/contact/views.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def get_initial(self):
 
         return initial
 
-    def get_success_url(self):
+    @staticmethod
```
Comment (unho) [inline pootle/apps/contact/views.py]: Please undo this since it is extending a class that didn't had this as a staticmethod.

**Comment by unho** [comment]
I have doubts about most of the changes here. I think that there are several methods that should be kept as regular ones so they correctly override methods from parent classes. The only one I don't it is a problem is the one in `pootle/core/search/backends/elasticsearch.py`.

**Comment by dwaynebailey** [comment]
Dropping, minimal value
